src/main/java/com/mongodb/kafka/connect/sink/writemodel/strategy/ReplaceOneBusinessKeyStrategy.java

```diff
@@ -56,4 +66,11 @@
               + " existing `_id` value in the business key then ensure `document.id.strategy.overwrite.existing=true`.");
     }
   }
+
+  @Override
+  public void configure(final MongoSinkTopicConfig configuration) {
```

Comment:
This is the important change - if using a partial id strategy, then we can't rely that the created id matches the full document.  So if that is the case we flatten the filter and use dot notation.

Reply:
Will there be an issue when the drivers allow for "." in field names in the future?

Reply:
Hopefully not, I'm not sure how the server will handle dot notation when querying field names with dots.

Reply:
It will treat it as a query for a nested document, e.g. 

    MongoDB Enterprise > db.test.insert({"a.b": 1})
    WriteResult({ "nInserted" : 1 })
    MongoDB Enterprise > db.test.find({"a.b" : 1})
    MongoDB Enterprise >